# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1445067667 -7200
#      Sat Oct 17 09:41:07 2015 +0200
# Branch stable
# Node ID 9e3b3aca6eadda647d45b32b51e905ff4b9a5846
# Parent  802dc52d4d469d38250acb6eb88e78e105a895cf
prevent doc browser from searching for an empty string (bug 46227)

* webinfo.cc (search): just return if search text is empty or only
  contains whitespace

diff --git a/libgui/src/qtinfo/webinfo.cc b/libgui/src/qtinfo/webinfo.cc
--- a/libgui/src/qtinfo/webinfo.cc
+++ b/libgui/src/qtinfo/webinfo.cc
@@ -236,16 +236,19 @@ webinfo::load_ref (const QString &ref_na
 
   if (_text_browser)
     _text_browser->setFocus();
 }
 
 void
 webinfo::search ()
 {
+  if (_search_line_edit->text ().trimmed ().isEmpty ())
+    return;   // do nothing if search field is empty or only has whitespaces
+
   if (_search_check_box->isChecked ())
     {
       // Global search
       QString results = _parser.global_search (_search_line_edit->text (), 5);
       _text_browser=addNewTab ("Results for: " + _search_line_edit->text ());
       _text_browser->setHtml (results);
     }
   else
